contracts/staking/cw20-stake/src/contract.rs

```diff
@@ -506,34 +473,33 @@ pub fn query_list_stakers(
 
 #[cfg_attr(not(feature = "library"), entry_point)]
 pub fn migrate(deps: DepsMut, _env: Env, msg: MigrateMsg) -> Result<Response, ContractError> {
-    // Set contract to version to latest
-    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
-
-    #[cw_serde]
-    struct BetaConfig {
-        pub admin: Addr,
-        pub token_address: Addr,
-        pub unstaking_duration: Option<Duration>,
-    }
+    use cw20_stake_v1 as v1;
 
+    let ContractVersion { version, .. } = get_contract_version(deps.storage)?;
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
     match msg {
-        MigrateMsg::FromBeta { manager } => {
```

Comment:
are there no more beta migrations left? 

i'm wondering if beta --> v2 is a concern

Reply:
there are none so far as i know. i did some musing on this and think that if you do want to do a DAO upgrade that spans multiple versions, the process will always be:

```
for each version:
  migrate(version)
```

we tag our releases such that doing this isn't too bad. if you need an intermediate code ID, you can always check out the tag and use docker to build a reproducible wasm.

otherwise, doing migrations across multiple versions is quite hard as you may have multiple layers of state changes to apply. for example, to migrate from beta to v2 on this contract you'd need to:

1. add the `manager` state key, do the migration to v1.
2. remove the `manager` state key, do the migration to v2.

this feels like a fair bit to keep track of, when we could require a little extra work during migration and keep these migrations to one-per-major-version.

does that sound reasonable to you?

Reply:
yeah, I think this is reasonable!